packages/hydrooj/src/model/problem.ts

```diff
@@ -431,11 +431,17 @@ export class ProblemModel {
         } else {
             throw new ValidationError('file', null, 'Invalid file');
         }
+        const summary = { readme: [], problems: [] };
         try {
             const problems = await fs.readdir(tmpdir, { withFileTypes: true });
             for (const p of problems) {
                 const i = p.name;
-                if (!p.isDirectory()) continue;
+                if (!p.isDirectory()) {
+                    if (p.name.toLowerCase().startsWith('readme')) {
+                        summary.readme.push(await fs.readFile(path.join(tmpdir, p.name), 'utf-8'));
```

Comment:
Chrome allows a maximum URL length of 8182 bytes. While a single Chinese character encodes to 9 chars in url, using querystring is not propriate in this case.

Reply:
Hmm... How does hydro handle this kind of temporary flashing message if query string is not suitable? Store in database or file store or session or somewhere else?